qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1963 @@
+
+        The qubits are assumed to be defined on physical qubits.
+
+        If ``lazy == True`` this circuit will extend a lazily constructed
+        quantum circuit. When an operation occurs that breaks the underlying
+        circuit scheduling assumptions such as adding a pulse instruction or
+        changing the alignment context the circuit will be
+        transpiled and scheduled into pulses with the current active settings.
+
+        Args:
+            gate: Gate to call.
+            qubits: Qubits to call gate on.
+            lazy: If false the circuit will be transpiled and pulse scheduled
+                immediately. Otherwise, it will extend the active lazy circuit
+                as defined above.
+        """
+        try:
+            iter(qubits)
+        except TypeError:
+            qubits = (qubits,)
+
+        qc = circuit.QuantumCircuit(self.num_qubits)
+        qc.append(gate, qargs=qubits)
+        self.call_circuit(qc, lazy=lazy)
+
+
+def build(backend=None,
+          schedule: Optional[Schedule] = None,
+          default_alignment: str = 'left',
+          default_transpiler_settings: Optional[Dict[str, Any]] = None,
+          default_circuit_scheduler_settings: Optional[Dict[str, Any]] = None
+          ) -> ContextManager[Schedule]:
+    """A context manager for launching the imperative pulse builder DSL.
+
+    To enter a building context and starting building a pulse program:
+
+    .. code-block:: python
+        :emphasize-lines: 9
+
+        from qiskit import execute
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+
+        d0 = pulse.DriveChannel(0)
+
+        with pulse.build() as pulse_prog:
+            pulse.play(pulse.Constant(100,0.5), d0)
+
+        # execute on real backend
+        # qiskit.execute(pulse_prog, backend)
+
+    Args:
+        backend (BaseBackend): A Qiskit backend. If not supplied certain
+            builder functionality will be unavailable.
+        schedule: a *mutable* pulse Schedule in which your pulse program will
+            be built.
+        default_alignment: Default scheduling alignment for builder.
+            One of ``left``, ``right``, ``sequential`` or an alignment
+            contextmanager.
+        default_transpiler_settings: Default settings for the transpiler.
+        default_circuit_scheduler_settings: Default settings for the
+            circuit to pulse scheduler.
+
+    Returns:
+        A new builder context which has the active builder inititalized.
+    """
+    return _PulseBuilder(
+        backend=backend,
+        schedule=schedule,
+        default_alignment=default_alignment,
+        default_transpiler_settings=default_transpiler_settings,
+        default_circuit_scheduler_settings=default_circuit_scheduler_settings)
+
+
+# Builder Utilities ############################################################
```

Comment:
```suggestion
# Builder Utilities
```
We shouldn't use these

Reply:
Fair enough 😄. Fixed all such instances in f3dac1893